### DISCUSSION TITLE
Which type of antialiasing is the internal d3d9 wrapper using?

### DISCUSSION THREAD

**Discussion by eddmanx** [elishacloud/dxwrapper#230] [Q&A] [answered]
In the guide it says **"This will force the built-in Direct3D9 AntiAliasing to be enabled"**, but which type is it? Does it have multiple levels?

**Comment by elishacloud** [comment]
Direct3D9 uses MSAA.  See here:  https://learn.microsoft.com/en-us/windows/win32/direct3d9/full-scene-antialiasing

**Reply by eddmanx** [reply]
Shouldn't MSAA have levels though, 2x, 4x, etc? It seems the ini only accepts 1.

**Reply by elishacloud** [reply, marked as answer]
Yes.  For the `AntiAliasing` setting in the dxwrapper.ini file you can use options 1, 2, 4, 8, and 16.

If you chose 1 then it will set it to the highest value supported by your video card.  If you chose a number larger than your video card supports then it will also set this to the highest your video card supports.  If you set this to 2 or 4 or some number smaller than your video card will support then it will set it to that value.

You can see this by the log file.  I just tested setting this to 2 and it worked, see log entry:

```log
8196 08:46:49.040 Setting MultiSample 2 Quality 0
```

**Reply by eddmanx** [reply]
Ah, I see, thanks. I got confused since the guide only mentions 0 and 1, and doesn't say 1 is auto.

**Reply by elishacloud** [reply]
I just updated the wiki with this information.  I hope it helps.
